fedimint-client-wasm/src/lib.rs

```diff
@@ -0,0 +1,80 @@
+use std::{str::FromStr, sync::Arc};
+
+use anyhow::format_err;
+use fedimint_client::{
+    module::IClientModule,
+    secret::{PlainRootSecretStrategy, RootSecretStrategy},
+    ClientHandleArc,
+};
+use fedimint_core::{invite_code::InviteCode, secp256k1::rand::thread_rng};
+use fedimint_ln_client::{LightningClientInit, LightningClientModule};
+use fedimint_mint_client::MintClientInit;
+use futures::StreamExt;
+use serde_json::json;
+use wasm_bindgen::prelude::*;
+
+#[wasm_bindgen]
+pub struct WasmClient {
+    client: ClientHandleArc,
+}
+
+#[wasm_bindgen]
+impl WasmClient {
+    #[wasm_bindgen]
+    pub async fn open() -> Result<Option<WasmClient>, JsError> {
+        Ok(None)
+    }
+
+    #[wasm_bindgen]
+    pub async fn join_federation(invite_code: String) -> Result<WasmClient, JsError> {
+        Self::join_federation_inner(invite_code)
+            .await
+            .map_err(|x| JsError::new(&x.to_string()))
+    }
+
+    async fn join_federation_inner(invite_code: String) -> anyhow::Result<WasmClient> {
+        let db = fedimint_core::db::mem_impl::MemDatabase::new();
+        let secret = fedimint_client::secret::PlainRootSecretStrategy::random(&mut thread_rng());
+        let root_secret = PlainRootSecretStrategy::to_root_secret(&secret);
+        let mut builder = fedimint_client::Client::builder(db.into());
+        builder.with_module(MintClientInit);
+        builder.with_module(LightningClientInit::default());
+        builder.with_primary_module(1);
+        let invite_code = InviteCode::from_str(&invite_code)?;
+        let config = fedimint_api_client::download_from_invite_code(&invite_code).await?;
+        let client = Arc::new(builder.join(root_secret, config, None).await?);
+        Ok(Self { client })
+    }
+
+    #[wasm_bindgen]
+    pub async fn rpc(&self, module: &str, body: String, cb: &js_sys::Function) {
+        let request = serde_json::from_str(&body).unwrap();
+        let mut stream = match module {
```

Comment:
I'm very confused about everything in this file, but I trust you know what you're doing. :D